src/plugins/explorer/sync-transactions.js

```diff
@@ -88,7 +91,41 @@ function createSyncer (config, eventBus, web3, queue, eventsRegistry, indexer) {
       })
   }
 
-  const getPastEvents = (fromBlock, toBlock, address) =>
+  function getPastEventsWithChunks (options) {
+    const CHUNK_SIZE = 4000
+    const { address, contract, eventName, fromBlock, toBlock, filter, metaParser, minBlock = 0, onProgress = noop } = options
+    let chunkIndex = 0
+    return pWhilst(
+      () => (fromBlock + CHUNK_SIZE * chunkIndex) < toBlock,
+      function () {
+        const newFromBlock = fromBlock + CHUNK_SIZE * chunkIndex
+        const newToBlock = Math.min(fromBlock + CHUNK_SIZE * (chunkIndex + 1), toBlock)
```

Comment:
`newToBlock` seems to have an off-by-one error. The sync should go from `fromBlock + CHUNK_SIZE * chunkIndex` to `fromBlock + CHUNK_SIZE * (chunkIndex + 1) - 1` instead.

Reply:
Thanks for catching this one! Fixed here https://github.com/autonomoussoftware/metronome-wallet-core/pull/32/commits/d210679451505871c1f651ee8771624fe4ceb25d